### PR TITLE
Fixed some broken tests and began to write IdentityComparer testing c…

### PR DESCRIPTION
**Pull request by mafurman**
Addresses #674

### CODE REVIEW

File: test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
+﻿using Microsoft.IdentityModel.Tokens.Saml;
```
Comment (mafurman) [inline test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs]: Done!

File: test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+using Xunit;
+using System.Linq;
+
+namespace Microsoft.IdentityModel.Tests
+{
+    // Checks to make sure that Claims, ClaimsIdentities, and ClaimsPrincipals with different properties are found 
+    // to be unequal by the IdentityComparer.AreEqual() method.
+    public class IdentityComparerTests
+    {
+
```
Comment (mafurman) [inline test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs]: Done!

File: test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+using System.Linq;
+
+namespace Microsoft.IdentityModel.Tests
+{
+    // Checks to make sure that Claims, ClaimsIdentities, and ClaimsPrincipals with different properties are found 
+    // to be unequal by the IdentityComparer.AreEqual() method.
+    public class IdentityComparerTests
+    {
+        
+        [Fact]//(Skip = "not fully functional yet")]
```
Comment (mafurman) [inline test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs]: Done! Sorry, forgot to remove that after I fixed the tests!

File: test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        public void CompareClaimsTest()
+        {
+            // Base claim that all tests will compare against.
+            var originalClaim = new Claim(ClaimTypes.Country, Default.Country, ClaimValueTypes.String, Default.Issuer, Default.OriginalIssuer);
+
+            var claimsToCompare = new List<Claim>()
+            {    
+            // Claim with different value for 'type'
+            new Claim(Guid.NewGuid().ToString(), Default.Country, ClaimValueTypes.String, Default.Issuer, Default.OriginalIssuer),
```
Comment (mafurman) [inline test/Microsoft.IdentityModel.Tokens.Tests/IdentityComparerTests.cs]: Done!

File: test/Microsoft.IdentityModel.Tests/IdentityComparer.cs [hunk 5]
```diff
@@ -689,7 +726,7 @@ public static bool AreWsFederationMessagesEqual(WsFederationMessage message1, Ws
         public static string BuildStringDiff(string label, object str1, object str2)
         {
             return (label ?? "label") + ": '" + GetString(str1) + "', '" + GetString(str2) + "'";
-        }
+        }
```
Comment (mafurman) [inline test/Microsoft.IdentityModel.Tests/IdentityComparer.cs]: Done! (not really sure how that happened)

**Comment by brentschmaltz** [comment]
:shipit:

**Pull request merged**